### PR TITLE
ST: Add transactional producer to system test

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

After https://github.com/strimzi/client-examples/pull/43 we are able to use transactional producer in our tests. This PR replace classic producer (which is used during upgrade continuously anyway) with transactional in test where we invoke rolling update manually. I will be happy to add it to other places, we just need to agree where it make sense.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=AlternativeReconcileTriggersST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=AlternativeReconcileTriggersST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=AlternativeReconcileTriggersST test_only

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java [hunk 1]
```diff
@@ -50,47 +50,51 @@
     @Test
     void testManualTriggeringRollingUpdate() {
         String topicName = KafkaTopicUtils.generateRandomNameOfTopic();
+        String clusterName = CLUSTER_NAME + "-manual-rolling";
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java:53]: It's only a question, but why do we need to rename it? Was there some issue?

**Review by im-konge — APPROVED** [review]
LGTM, just a question, because I'm curious.

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java:53]: Yes, I think the second test get same PV (or something like that) and CO wasn't able to roll some pods.

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java:53]: That make sense :) thanks for explanation.

**Review by tombentley — COMMENTED** [review]
I've not really looked at this test before, but it seems that while it's able to test that it's possible to send some number of messages before and after the roll, it doesn't test that it's possible to sent messages continuously during the roll. It needs to. It should also check (which it may do already, I didn't look) that there are no errors in trying to begin or end transactions, or send messages (in particular timeouts). Or equivalently we need an asserion on the max time during the roll that the producer is waiting for:
    * message ack
    * transaction begin and commit

**Comment by Frawless** [comment]
@tombentley actually the test creates two jobs, one for producer and one for consumer. Producer sends messages each second, together 500 messages which means the test will take 500s+. Each 10th message producer commits transactions. Ack is set `acsk=all` .

**Comment by Frawless** [comment]
We use `client-examples` images and after https://github.com/strimzi/client-examples/pull/43 we are able to use producer with transactions.

**Comment by tombentley** [comment]
@Frawless, but does it fail the test if the producer has a timeout? Or could there be a bunch of timeouts during  the test and then producer recovers and finishes sending the 500 requests?

**Comment by Frawless** [comment]
If consumer didn't receive 500 messages, the test fail. I was able to see some timeouts in producer logs, but all not acked messages were resent, because consumer consumed all messages.

**Comment by tombentley** [comment]
But the producer should not have a timeout. It doesn't matter whether it was able to recover or not. Having a producer stall for 2 minutes before getting a timeout and then recovering from it is still a problem.

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=AlternativeReconcileTriggersST test_only

**Review by tombentley — APPROVED** [review]

**Pull request merged**